pkg/cmd/util/clientcmd/factory.go

```diff
@@ -291,19 +298,14 @@ func (c *clientCache) ClientConfigForVersion(version string) (*kclient.Config, e
 		// negotiation logic should be refactored upstream to support downstream
 		// reuse so that we don't need to do any of this cache or negotiation
 		// duplication.
-		client.SetOpenShiftDefaults(config)
-		c.defaultConfig = config
-		negotiatingClient, err := kclient.New(config)
+		negotiatingConfig := *c.defaultConfig
+		client.SetOpenShiftDefaults(&negotiatingConfig)
```

Comment:
I wanted to SetOpenShiftDefaults on a copy of the defaultConfig. Previously, defaultConfig didn't have SetOpenShiftDefaults applied to it, only to copies of it... want to preserve that behavior